dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ReplaceZeroRegisters.java

```diff
@@ -0,0 +1,50 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.expression.ExpressionFactory;
+import com.dat3m.dartagnan.expression.type.IntegerType;
+import com.dat3m.dartagnan.program.Function;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.RegReader;
+import com.dat3m.dartagnan.program.event.RegWriter;
+import com.dat3m.dartagnan.program.event.core.Local;
+
+/*
+    This pass replaces every RegWritter
+        zr <- e
+    targeting a zero register to
+        dummy <- e; zr <- 0
+    and for every RegReader reading zr, it inserts an event zr <- 0 before it
+*/
+public class ReplaceZeroRegisters implements FunctionProcessor {
+
+    public static ReplaceZeroRegisters newInstance() {
+        return new ReplaceZeroRegisters();
+    }
+
+    @Override
+    public void run(Function function) {
+
+        for (RegWriter rw : function.getEvents(RegWriter.class)) {
+            if (rw.getResultRegister().isZeroRegister()) {
+                Register zr = rw.getResultRegister();
+                Register dummy = rw.getThread().newRegister(zr.getType());
+                rw.setResultRegister(dummy);
+                rw.insertAfter(EventFactory.newLocal(zr,
+                        ExpressionFactory.getInstance().makeZero((IntegerType) zr.getType())));
```

Comment:
I think the replacement register should have a name to understand its intention, e.g., `__zeroRegDummy_<origRegName>` or `__ignored`.
Also, you can probably reuse a single replacement register (per function) since all usages should be type-consistent.
Of course, this assumes that you change the pass to take as input the replacement register (or rather its name), then it can perform per function a lookup on that register name, construct a same-type dummy and do the replacement.